catroid/src/main/java/org/catrobat/catroid/utils/ImportProjectsFromExternalStorage.java

```diff
@@ -0,0 +1,169 @@
+
+import static org.catrobat.catroid.common.SharedPreferenceKeys.SHOW_COPY_PROJECTS_FROM_EXTERNAL_STORAGE_DIALOG;
+import static org.catrobat.catroid.io.StorageOperations.transferData;
+
+public class ImportProjectsFromExternalStorage implements DialogInterface.OnClickListener {
+	public static final String TAG = ImportProjectsFromExternalStorage.class.getSimpleName();
+
+	private static final File POCKET_CODE_ROOT_DIRECTORY = new File(
+			CatroidApplication.getAppContext().getFilesDir().getAbsolutePath() + "/Programs");
+	private static final File EXTERNAL_STORAGE_ROOT_DIRECTORY = new File(
+			Environment.getExternalStorageDirectory().getAbsolutePath() + "/Pocket Code");
+	private Activity activity;
+
+	public ImportProjectsFromExternalStorage(Activity mActivity) {
+		activity = mActivity;
+	}
+
+	private static File copyProjectsFromExternalToInternalStorage(File srcDir, File dstDir) throws IOException {
+		UniqueNameProvider uniqueNameProvider = new UniqueNameProvider();
+		List<String> scope = new ArrayList<>();
+		if (!srcDir.exists()) {
+			throw new FileNotFoundException("Directory: " + srcDir.getAbsolutePath() + " does not exist.");
+		}
+
+		if (!srcDir.isDirectory()) {
+			throw new IOException(srcDir.getAbsolutePath() + " is not a directory.");
+		}
+		if (!dstDir.exists()) {
+			dstDir.mkdir();
+		}
+
+		if (!dstDir.isDirectory()) {
+			throw new IOException("Cannot create directory: " + dstDir.getAbsolutePath());
+		}
+
+		for (File srcFile : srcDir.listFiles()) {
+			String srcFileName = srcFile.getName();
+			if (srcFile.isDirectory()) {
+				boolean fileExists = false;
+				for (File dstFile : dstDir.listFiles()) {
+					if (dstFile.getName().equals(srcFileName) && dstFile.getParent().endsWith("Programs") && srcFile.getParent().endsWith("Pocket Code") && !srcFileName.equals("backpack")) {
+						fileExists = true;
+						scope.add(srcFileName);
+						break;
+					}
+				}
+				if (fileExists) {
+					String newFileName = uniqueNameProvider.getUniqueName(srcFileName, scope);
+					copyProjectsFromExternalToInternalStorage(srcFile, new File(dstDir, newFileName));
+				} else {
+					copyProjectsFromExternalToInternalStorage(srcFile, new File(dstDir, srcFileName));
+				}
+			} else {
+				transferData(srcFile, new File(dstDir, srcFileName));
+			}
+		}
+
+		return dstDir;
+	}
+
+	public void importOldPocketCodePrograms() {
+		if (FlavoredConstants.DEFAULT_ROOT_DIRECTORY.equals(POCKET_CODE_ROOT_DIRECTORY) && pocketCodeExternalStorageDirectoryExists()) {
+			showImportProjectsDialog(activity);
+		}
+	}
+
+	private boolean pocketCodeExternalStorageDirectoryExists() {
+		return EXTERNAL_STORAGE_ROOT_DIRECTORY.exists();
+	}
+
+	private void showImportProjectsDialog(Activity activity) {
+		AlertDialog.Builder alertDialog = new AlertDialog.Builder(activity);
+		alertDialog.setTitle(R.string.import_dialog_title)
+				.setCancelable(false)
+				.setMessage(R.string.import_dialog_message)
+				.setPositiveButton(R.string.ok, this)
```

Comment:
give the user some information that "ok" means "move" and "import_dialog_copy_only" means "copy"

Reply:
where should be the infos ? in the dialog message?